### PR TITLE
Update ORT web page documents to install CUDA 12

### PR DESCRIPTION
**Pull request by jchen351**
### Description

This pull request introduces changes to the `docs/execution-providers/CUDA-ExecutionProvider.md` file, which provides documentation for the CUDA Execution Provider. The changes mainly involve reformatting the text to improve readability and updating the content to reflect the latest changes in the CUDA Execution Provider.

* Formatting and Readability Improvements:
  * The text has been reformatted to improve readability, with long lines of text broken down into shorter lines. [[1]](diffhunk://#diff-c0b7353d9f98e4bba18b2e9c915bc837eb69846c55ed0e8e0b7314119690e144R10-R41) [[2]](diffhunk://#diff-c0b7353d9f98e4bba18b2e9c915bc837eb69846c55ed0e8e0b7314119690e144L47-R118) [[3]](diffhunk://#diff-c0b7353d9f98e4bba18b2e9c915bc837eb69846c55ed0e8e0b7314119690e144R127-R163) [[4]](diffhunk://#diff-c0b7353d9f98e4bba18b2e9c915bc837eb69846c55ed0e8e0b7314119690e144R172-R175) [[5]](diffhunk://#diff-c0b7353d9f98e4bba18b2e9c915bc837eb69846c55ed0e8e0b7314119690e144L153-R233) [[6]](diffhunk://#diff-c0b7353d9f98e4bba18b2e9c915bc837eb69846c55ed0e8e0b7314119690e144L185) [[7]](diffhunk://#diff-c0b7353d9f98e4bba18b2e9c915bc837eb69846c55ed0e8e0b7314119690e144L215-R279) [[8]](diffhunk://#diff-c0b7353d9f98e4bba18b2e9c915bc837eb69846c55ed0e8e0b7314119690e144L256-R320) [[9]](diffhunk://#diff-c0b7353d9f98e4bba18b2e9c915bc837eb69846c55ed0e8e0b7314119690e144L265-R348) [[10]](diffhunk://#diff-c0b7353d9f98e4bba18b2e9c915bc837eb69846c55ed0e8e0b7314119690e144L379)
  * Code snippets have been updated for better clarity and readability. [[1]](diffhunk://#diff-c0b7353d9f98e4bba18b2e9c915bc837eb69846c55ed0e8e0b7314119690e144R127-R163) [[2]](diffhunk://#diff-c0b7353d9f98e4bba18b2e9c915bc837eb69846c55ed0e8e0b7314119690e144R172-R175) [[3]](diffhunk://#diff-c0b7353d9f98e4bba18b2e9c915bc837eb69846c55ed0e8e0b7314119690e144L153-R233)

* Content Updates:
  * The CUDA Execution Provider's documentation has been updated to reflect the latest changes in the provider. This includes updates to the CUDA Execution Provider's configuration options and performance tuning recommendations. [[1]](diffhunk://#diff-c0b7353d9f98e4bba18b2e9c915bc837eb69846c55ed0e8e0b7314119690e144R10-R41) [[2]](diffhunk://#diff-c0b7353d9f98e4bba18b2e9c915bc837eb69846c55ed0e8e0b7314119690e144L47-R118) [[3]](diffhunk://#diff-c0b7353d9f98e4bba18b2e9c915bc837eb69846c55ed0e8e0b7314119690e144R127-R163) [[4]](diffhunk://#diff-c0b7353d9f98e4bba18b2e9c915bc837eb69846c55ed0e8e0b7314119690e144R172-R175) [[5]](diffhunk://#diff-c0b7353d9f98e4bba18b2e9c915bc837eb69846c55ed0e8e0b7314119690e144L153-R233) [[6]](diffhunk://#diff-c0b7353d9f98e4bba18b2e9c915bc837eb69846c55ed0e8e0b7314119690e144L185) [[7]](diffhunk://#diff-c0b7353d9f98e4bba18b2e9c915bc837eb69846c55ed0e8e0b7314119690e144L215-R279) [[8]](diffhunk://#diff-c0b7353d9f98e4bba18b2e9c915bc837eb69846c55ed0e8e0b7314119690e144L256-R320) [[9]](diffhunk://#diff-c0b7353d9f98e4bba18b2e9c915bc837eb69846c55ed0e8e0b7314119690e144L265-R348) [[10]](diffhunk://#diff-c0b7353d9f98e4bba18b2e9c915bc837eb69846c55ed0e8e0b7314119690e144L379)

* Removals:
  * Some unnecessary lines of text have been removed from the document. [[1]](diffhunk://#diff-c0b7353d9f98e4bba18b2e9c915bc837eb69846c55ed0e8e0b7314119690e144L215-R279) [[2]](diffhunk://#diff-c0b7353d9f98e4bba18b2e9c915bc837eb69846c55ed0e8e0b7314119690e144L379)

### CODE REVIEW

File: docs/install/index.md [hunk 1]
```diff
@@ -8,33 +8,78 @@ redirect_from: /docs/how-to/install
...
+Ensure you have installed the latest version of the Azure Artifacts keyring from the "Get the tools" menu.
+If you don't already have one, create a virtualenv
+using [these instructions](https://go.microsoft.com/fwlink/?linkid=2103878) from the official Python documentation.
+Per the instructions, "it is always recommended to use a virtualenv while developing Python applications."
+
+Add a pip.ini (Windows) or pip.conf (Linux) file to your virtualenv
+
+```text
+[global]
+index-url=https://aiinfra.pkgs.visualstudio.com/PublicPackages/_packaging/onnxruntime-cuda-12/pypi/simple/
```
Comment (tianleiwu) [inline docs/install/index.md:78]: why not just
```
pip install onnxruntime-gpu -i https://aiinfra.pkgs.visualstudio.com/PublicPackages/_packaging/onnxruntime-cuda-12/pypi/simple/
```

File: docs/execution-providers/CUDA-ExecutionProvider.md [hunk 2]
```diff
@@ -7,131 +7,160 @@ redirect_from: /docs/reference/execution-providers/CUDA-ExecutionProvider
...
+11.x version.
+Please
+reference [Nvidia CUDA Minor Version Compatibility](https://docs.nvidia.com/deploy/cuda-compatibility/#minor-version-compatibility).
+
+| ONNX Runtime             | CUDA   | cuDNN                                   | Notes                                                                                                                                                                                                                               |
+|--------------------------|--------|-----------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| 1.17                     | 12.2   | 8.9.2.26 (Linux)<br/>8.9.2.26 (Windows) | The default CUDA version for ORT 1.17 is CUDA 11.8 to install CUDA 12 package please look at [Install ORT](../install).<br>Due to low demend on Java GPU package, only C++/C# Nuget and Python packages are released with CUDA 12.2 |
```
Comment (tianleiwu) [inline docs/execution-providers/CUDA-ExecutionProvider.md:40]:
```suggestion
| 1.17                     | 12.2   | 8.9.2.26 (Linux)<br/>8.9.2.26 (Windows) | The default CUDA version for ORT 1.17. To install CUDA 12 package, please look at [Install ORT](../install).<br>Due to low demend on Java GPU package, only C++/C# Nuget and Python packages are released with CUDA 12.2 |
```

**Review by MaanavD — COMMENTED** [review]
Reads fine to me, but will have merge conflicts with #19355 , so let's get that one merged first before this one.

**Review by MaanavD — APPROVED** [review]
Approved based on discussion w/ Sophie.

**Pull request merged**